### ISSUE TITLE
[MSB6003] The specified task executable "" could not be run.

### ISSUE THREAD

**Issue by eliottrobson** [reinforced/Reinforced.Typings#106]
I have recently updated to version `1.5.0` and am now receiving the following error:

` Reinforced.Typings.targets(59, 3): [MSB6003] The specified task executable "" could not be run. No such file or directory`

Is there a known fix or anything I can do to get more information about what is happening?

**Comment by pavel-b-novikov** [comment]
Use `RtForceTargetFramework` in Reinforced.Typings.settings.xml

**Comment by eliottrobson** [comment]
I have added `<RtForceTargetFramework>netcoreapp2.1</RtForceTargetFramework>` but it has the same error.

**Comment by pavel-b-novikov** [comment]
Then I need to know more details about your environment

**Comment by eliottrobson** [comment]
I use MacOS and Rider IDE. I can post my RT config file if that helps?
I am using `.NET Core v2.1.6` and `SDK 2.1.500`

**Comment by pavel-b-novikov** [comment]
I do not have ability to test it on Mac, so it will make sense to contact via Skype/TV. 
Skype: nsu_the_cjay

**Comment by eliottrobson** [comment]
Okay, I will message you when I get home tonight and I'll do some more research today.

**Comment by pavel-b-novikov** [comment]
stayin' in touch

**Comment by eliottrobson** [comment]
I have also tried building using Visual Studio Mac and I get the same albeit more detailed error.

```/Users/eliottrobson/.nuget/packages/reinforced.typings/1.5.0/build/Reinforced.Typings.targets(3,3): Error MSB6003: The specified task executable "" could not be run. ApplicationName='dotnet.exe', CommandLine=' "/Users/eliottrobson/.nuget/packages/reinforced.typings/1.5.0/tools/netcoreapp2.1/rtcli.dll" profile "/var/folders/j5/xc_tyl5j1t97f3bdqhxc6vpm0000gn/T/tmp628a4eaf.tmp"', CurrentDirectory='', Native error= Cannot find the specified file (MSB6003) (App.Web)```

I'm not sure if that is more helpful? Is there anyway for me to see more detailed logs?

**Comment by pavel-b-novikov** [comment]
Well, it sounds like very specific nix-system problem

**Comment by eliottrobson** [comment]
I have done some more research, let me know what you think.

The only reference to `dotnet.exe` I can find is in the file `Reinforced.Typings/Reinforced.Typings.Integrate/RtCli.cs`
https://github.com/reinforced/Reinforced.Typings/blob/32597244e129b2939872686101d8d239c957197a/Reinforced.Typings.Integrate/RtCli.cs

I can be fairly confident that `IsCore` is returning correctly because `rtcli.exe` is not referenced.

Therefore, I think that the issue lies in `#if NETCORE`
I have decompiled `reinforced.typings.1.5.0.nupkg\build\netstandard2.0\Reinforced.Typings.Integrate.dll`

and I see the following code

```
protected override string GenerateFullPathToTool()
{
    if (this.IsCore)
        return "dotnet.exe";
    return Path.Combine(this.GetProperRtcliPath(), this.ToolName);
}
```

This accurately maps to a false evaluation of the if directive when compiling for netstandard. I have not used if directives or custom constants before so I'm afraid I can't dig too much deeper but I hope this is useful to provide a fix.

**Comment by romansp** [comment]
Hi, I'm hitting exact same issue with 1.5.0 on MacOS. 
Minimal repro available here: https://github.com/romansp/reinforced-task-not-found
It works fine when running on Windows.

<details>
<summary>Build Log</summary>
<pre>
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 35.45 ms for /Users/roman.pavlov/Documents/dev/reinforced/reinforced.csproj.
/Users/roman.pavlov/.nuget/packages/reinforced.typings/1.5.0/build/Reinforced.Typings.targets(59,3): error MSB6003: The specified task executable "" could notbe run. No such file or directory [/Users/roman.pavlov/Documents/dev/reinforced/reinforced.csproj]

Build FAILED.

/Users/roman.pavlov/.nuget/packages/reinforced.typings/1.5.0/build/Reinforced.Typings.targets(59,3): error MSB6003: The specified task executable "" could notbe run. No such file or directory [/Users/roman.pavlov/Documents/dev/reinforced/reinforced.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.08
</pre>
</details>

Let me know how I can help.

**Comment by pavel-b-novikov** [comment]
> I have checked the Reinforced.Typings/cake/build.cake file and I notice that the IntegrateProject does NOT have a corresponding RtAdditionalConstants section like the other files do.

Holy fck... You're right

**Comment by eliottrobson** [comment]
I tried to test if adding that would work but couldn't setup my environment correctly.
This is the tool I used on Windows (free) to decompile the code. [https://www.jetbrains.com/decompiler/](https://www.jetbrains.com/decompiler/). If you need to check that the generated code is correct it's a nice easy way to see what the actual output is.

**Comment by pavel-b-novikov** [comment]
Thanks. I like JetBrains products and use dotPeek since it 1st version. Tomorrow I will revise task logic and update version

**Comment by pavel-b-novikov** [comment]
Fixed in version 1.5.1. Tested on VM with Ubuntu 18.10
